bootloader/test_infra/src/main.rs

```diff
@@ -0,0 +1,181 @@
+use crate::{test_count_tracer::TestCountTracer, tracer::BootloaderTestTracer};
+use colored::Colorize;
+use once_cell::sync::OnceCell;
+use std::process;
+use std::{env, sync::Arc};
+use tracing_subscriber::fmt;
+use tracing_subscriber::prelude::__tracing_subscriber_SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
+use vm::{
+    HistoryDisabled, L1BatchEnv, L2BlockEnv, SystemEnv, TxExecutionMode, Vm, VmExecutionMode,
+    VmTracer,
+};
+use zksync_contracts::{
+    read_zbin_bytecode, BaseSystemContracts, ContractLanguage, SystemContractCode,
+    SystemContractsRepo,
+};
+use zksync_state::{
+    InMemoryStorage, StoragePtr, StorageView, IN_MEMORY_STORAGE_DEFAULT_NETWORK_ID,
+};
+use zksync_types::system_contracts::get_system_smart_contracts_from_dir;
+use zksync_types::{block::legacy_miniblock_hash, Address, L1BatchNumber, MiniblockNumber, U256};
+use zksync_types::{L2ChainId, Transaction};
+use zksync_utils::bytecode::hash_bytecode;
+use zksync_utils::{bytes_to_be_words, u256_to_h256};
+
+mod hook;
+mod test_count_tracer;
+mod tracer;
+
+/// Executes the "internal transfer test" of the bootloader -- the test that
+/// returns the amount of gas needed to perform and internal transfer, assuming no gas price
+/// per pubdata, i.e. under assumption that the refund will not touch any new slots.
+fn execute_internal_bootloader_test() {
```

Comment:
are you sure the name and the comment are correct? It seems like actually just the whole test suit is executed

Reply:
ups - I've copied it from somewhere and forgot to fix. thanks.